src/GLua/AG/PrettyPrint.ag

```diff
@@ -182,16 +186,26 @@ infixl 6 <->
 a <-> b | a == empty = b
         | b == empty = a
         | otherwise  = a <> zchr ' ' <> b
+
+-- Always 10, so there is a default for precedence
+infixr 5 $$$
+($$$) :: Int -> Int -> Int
+a $$$ b = if a > b then 10 else 10
+
 }
 
 
 attr AllStuff
     syn pretty :: Doc
     inh indent :: Int
+    inh neighborPrecedence :: Int
```

Comment:
I'm having some trouble understanding what "neighbor" here means. From the code it seems that it transfers the precedence of the operator that an expression is affected by, but I'm not sure how the word "neighbor" is related to it. 

Also, this is passed through the entire AST through `AllStuff`. That seems odd, since the parentheses are only reasoning about expressions. The scope of this inherited variable can probably be limited.

Reply:
Yeah, your description is correct. Maybe `requiredPrecedence` would be a better name. And yeah the scope should probably be changed, I think I just didn't know how to set the default value for it...